### PR TITLE
feat: Add pydantic config models

### PR DESCRIPTION
**Pull request by robinholzi**
To improve typing within the codebase w.r.t. to the config objects, I transitioned the `jsonschema`s to pydantic models.

Those model will make the supervisor a lot more concise as all the validation logic will be stripped out and arguments will have proper typing. As a result it will be a lot easier to have uniform benchmarking across several pipeline steps without significant code duplication (needed for policy evaluation)

### CODE REVIEW

File: modyn/storage/src/CMakeLists.txt [hunk 1]
```diff
@@ -135,4 +135,4 @@ target_compile_definitions(modyn-storage-library PUBLIC ${MODYN_COMPILE_DEFINITI
 set(INCLUDE_EXPR "$<TARGET_PROPERTY:modyn-storage-library,INCLUDE_DIRECTORIES>")
 set(INCLUDE_FILTER "$<FILTER:${INCLUDE_EXPR},INCLUDE,.*/.*>")
 set(INCLUDE_JOINED "$<JOIN:${INCLUDE_FILTER},,>")
-target_compile_definitions(modyn-storage-library PRIVATE "INCLUDE_DIRECTORIES=\"${INCLUDE_JOINED}\"")
```
Comment (MaxiBoether) [inline modyn/storage/src/CMakeLists.txt:138]: let's not touch this file with this commit :)

**Review by MaxiBoether — APPROVED** [review]
I validated the pipeline config, client config and an excerpt of the modyn config. Since the tests also run through, I think this is good to go. Thank you!

I think your IDE might be using different settings from our flake/black config since this PR introduces many (unrelated) linebreaks (but to me, that's ok, can merge them) :-)

File: modyn/storage/src/CMakeLists.txt (hunk 1 above)
Comment (robinholzi) [reply modyn/storage/src/CMakeLists.txt]: Ok, I reverted these changes. Going forward, would you prefer file endings with or without newlines?

**Comment by XianzheMa** [comment]
Großartig! Thank you for this pain saver!

**Pull request merged**